### PR TITLE
added prims algorithm issue-3304

### PR DESCRIPTION
**Pull request by kavyasree-123**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
yes i have

### Description
prims algorithm is being added to cpp ds files
Prim's Algorithm is used to find the minimum spanning tree from a graph. Prim's algorithm finds the subset of edges that includes every vertex of the graph such that the sum of the weights of the edges can be minimized.
time_complexity_O(ElogV) where v no of vertices and e no of edges

### Checklist

- [yes ] I've read the contribution guidelines.
- [yes ] I've checked the issue list before deciding what to submit.
- [ yes] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

[#3304](https://github.com/TesseractCoding/NeoAlgo/issues/3304)  added code to this issue please kindly verify

### CODE REVIEW

**Comment by kavyasree-123** [comment]
please review my pr and merge prim's algorithm
@bhav09 , @siddharth25pandey @HarshCasper , @ankitaggarwal23

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Few changes: 

- Add Multi-Line Comments 
- Add a beginner-friendly description at the top
- Use proper code Format 
- Remove extra lines of code and spaces
- Add sample I/O at the end of the Code
- Add Time and Space Complexity at the end of the Code

**Comment by kavyasree-123** [comment]
I have added multi line commands added description and all the others are done. please review

File: C-Plus-Plus/ds/prims algo.cpp [hunk 1]
```diff
@@ -0,0 +1,79 @@
+/*Prim's algorithm is a minimum spanning tree algorithm that takes a graph as input and finds 
+the subset of the edges of that graph which. form a tree that includes every vertex. 
+has the minimum sum of weights among all the trees that can be formed from the graph.
+(issue-3304)*/
+#include <iostream>
+#include <cstring>
```
Comment (siddharth25pandey) [inline C-Plus-Plus/ds/prims algo.cpp]: Write single header file **bits/stdc**

File: C-Plus-Plus/ds/prims algo.cpp [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  {5, 7, 85, 2, 1, 58, 45,1,7},
+  {54 , 0, 75, 9, 2, 46 ,17,4,5},
+  {63, 16, 0, 41, 66 , 48,36,48,84},
+  {7, 9, 5, 0,65, 1,3,87,59 },
+  {6, 72, 76, 75,48,95,49,46,47},
+  {19, 25,65,39,17,24,28,12,14},
+  {76,89,94,15,14,28,27,78,74},
+  {1,75,48,75,48,63,15,4,8},
+  {3,37,36,18,25,2,9,29,43}
+};
```
Comment (siddharth25pandey) [inline C-Plus-Plus/ds/prims algo.cpp]: Take dynamic input from the user.

File: C-Plus-Plus/ds/prims algo.cpp [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+/* in main we give inputs to graph array and calls operation function */ 
+int main() {
+  int all_edges;
+   int array[number];
+/*memset Sets the first num bytes of the block of memory 
+pointed by ptr to the specified value faster than loop*/
+  memset(array, false, sizeof(array));
+  all_edges = 0;
+  array[0] = true;
+  int row , coloumn;
```
Comment (siddharth25pandey) [inline C-Plus-Plus/ds/prims algo.cpp]: Display message, what are you taking input from the user

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@kavyasree-123 Do the suggested changes and update the **Readme** file changes

File: C-Plus-Plus/ds/prims algo.cpp (hunk 3 above)
Comment (kavyasree-123) [reply C-Plus-Plus/ds/prims algo.cpp]: done i have described

File: C-Plus-Plus/ds/prims algo.cpp (hunk 2 above)
Comment (kavyasree-123) [reply C-Plus-Plus/ds/prims algo.cpp]: yes i have changed the code took input from user

File: C-Plus-Plus/ds/prims algo.cpp (hunk 1 above)
Comment (kavyasree-123) [reply C-Plus-Plus/ds/prims algo.cpp]: yes i have changed the header file

**Comment by kavyasree-123** [comment]
@siddharth25pandey i have made the changes and updated the code please verify

**Comment by siddharth25pandey** [comment]
@kavyasree-123 Please check carefully, what I wrote earlier. Have a look [here](https://github.com/TesseractCoding/NeoAlgo/pull/3542#pullrequestreview-616049464)+
**Please update the Readme file details**

**Comment by kavyasree-123** [comment]
hello i have updated readme file please verify @siddharth25pandey

**Review by anushkrishnav — APPROVED** [review]

**Comment by anushkrishnav** [comment]
@siddharth25pandey Looks good to me

**Review by siddharth25pandey — APPROVED** [review]
@kavyasree-123 Able to merge, Thanks!

**Comment by siddharth25pandey** [comment]
@all-contributors please add @kavyasree-123 for her code

**Comment by kavyasree-123** [comment]
thank you so much for all mentors..

**Pull request merged**